src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTests.cs

```diff
@@ -6952,5 +6952,54 @@ .maxstack  2
   IL_0011:  ret
 }");
         }
+
+        [Fact]
+        public void EvaluateField_PropertyUsesField()
+        {
+            var source =
+@"class C
+{
+    public int P
+    {
+        get
+        {
+            field = 10;
+            return field;
+        }
+        set => field = value;
+    }
+}";
+            var compilation = CreateCompilation(source, options: TestOptions.DebugDll);
+            compilation.VerifyDiagnostics();
+            WithRuntimeInstance(compilation, runtime =>
+            {
+                var context = CreateMethodContext(runtime, "C.get_P");
+                var testData = new CompilationTestData();
+                var result = context.CompileExpression("field", out var error, testData);
+
+                // PROTOTYPE(semi-auto-props): Likely incorrect behavior.
```

Comment:
@AlekseyTs Can you confirm please?

Reply:
So looking into this, it's failing because we get a `PEMethod` while binder checks for `ContainingMemberOrLambda is SourcePropertyAccessorSymbol`. I can add a check for the `PEMethod` as well, but I don't see anywhere the binder do that, so I'm feeling this is not the right way to do things. Any pointers @AlekseyTs ? Thanks!

Reply:
I attempted to fix it in 86b7d89, not sure it's the right way to do it.